src/Tools/dotnet-monitor/HostBuilder/HostBuilderSettings.cs

```diff
@@ -42,6 +55,22 @@ internal sealed class HostBuilderSettings
                     Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.UserProfile), "." + ProductFolderName) :
                     Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.ApplicationData), ProductFolderName));
 
+        // Location where extensions are stored by default.
+        // Windows: "%USERPROFILE%\.dotnet\Tools"
+        // Other: "%XDG_CONFIG_HOME%/.dotnet/tools" OR "%HOME%/.dotnet/tools" -> THIS HAS NOT BEEN TESTED YET ON LINUX
+        public static readonly string DotnetToolsExtensionDirectoryPath =
+            GetEnvironmentOverrideOrValue(
+                DotnetToolsExtensionDirectoryOverrideEnvironmentVariable,
+                RuntimeInformation.IsOSPlatform(OSPlatform.Windows) ?
+                    Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.UserProfile), "." + DotnetFolderName, ToolsFolderName) :
+                    Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.ApplicationData), "." + DotnetFolderName, ToolsFolderName));
+
+        // Location for dotnet-monitor's executing assembly.
+        public static readonly string ExecutingAssemblyDirectoryPath =
```

Comment:
I don't think this override is necessary: since "next to me", the local user settings, and the shared settings extension discoverers are the same implementation, you can choose to drop the extension in the already existing directory specified by DotnetMonitorTestSettings__UserConfigDirectoryOverride; testing this is the equivalent of testing any of the others that use the same implementation.

Reply:
I wasn't sure exactly what you wanted here, so I took my best guess and ripped out the functionality surrounding setting this via environment variables since it wasn't needed - I can change this if you want something else.